projects/find/find.py

```diff
@@ -0,0 +1,37 @@
+import os
+import argparse
+
+"""This method is created to find all files and directories, given a specific path. Optionally, your can use -name flag
+to search only the files that match to a specific name"""
+
+
+def find(path, name=None):
+    if not os.path.exists(path):
+        return "Path doesn't exist or wrong: please try again"
+    stack = []
+    stack.append(path)
+    while len(stack) > 0:
+        stack_element = stack.pop(0)
+        if not os.path.islink(stack_element):
+            if os.path.isdir(stack_element):
+                adj_list = os.listdir(stack_element)
+                adj_list.sort(reverse=True)
+                for adj in adj_list:
+                    stack.insert(0, os.path.join(stack_element, adj))
```

Comment:
this is quite inefficient: inserting at position 0 will make all other elements shift by 1 and you this fore each element in adj_list. can you avoid this and still use a stack?